src/tblib/__init__.py

```diff
@@ -19,30 +19,24 @@
 
 
 class _AttrDict(dict):
+    __slots__ = ()
+
     def __getattr__(self, attr):
         return self[attr]
 
 
-class __traceback_maker(Exception):
-    pass
-
-
 class Code(object):
     def __init__(self, code):
         self.co_filename = code.co_filename
         self.co_name = code.co_name
-        self.co_nlocals = code.co_nlocals
-        self.co_stacksize = code.co_stacksize
-        self.co_flags = code.co_flags
-        self.co_firstlineno = code.co_firstlineno
 
 
 class Frame(object):
     def __init__(self, frame):
         self.f_globals = dict([
-            (k, v)
-            for k, v in frame.f_globals.items()
-            if k in ("__file__", "__name__")
+            (k, frame.f_globals[k])
```

Comment:
Sure, I just figured that this was O(1) instead of O(n) on the number of items in the globals dict.